docs/src/tutorials/diffusion.jl

```diff
@@ -0,0 +1,273 @@
+φ_gauss = @. exp(-(lat^2 + long^2) / σ^2) * (z < 0.005);
+
+# Note how we multiplied by `(z < 0.005)` to ensure that the Gaussian
+# perturbation is only on the lowest face.
+#
+# When working with `ClimaCore`, `ClimaTimesteppers` requires all the evolved
+# variables to be packed in `ClimaCore.FieldVector`s. In this case, we only
+# have one variable, which we will call `my_var`
+#
+# Let us pack this into a a `FieldVector`
+Y₀ = ClimaCore.Fields.FieldVector(; my_var = copy(φ_gauss));
+
+# We copy `φ_gauss` because it it will be modified by the integrator.
+
+# ## The equations and the integrator
+
+# The diffusion equation for $u$ is $\partial_t u = K \nabla^2 u$, with $K$
+# diffusivity. To solve this equation with `ClimaTimesteppers`, we have to
+# provide the tendencies (ie, the right-hand-sides). We want to solve the
+# diffusion equation horizontally and vertically. Horizontally, we will treat
+# the evolution explicitly (currently, there is no other option for our spectral
+# elements). On the other hand, we will solve the vertical diffusion implicitly.
+# So, this problem has two tendencies, the explicit and the implicit ones.
+#
+# So, let us start by defining the explicit tendency.
+
+# A tendency for `ClimaTimesteppers.jl` is a function that generally takes four
+# arguments `(∂ₜY, Y, p, t)`. `∂ₜY` is the right-hand side, which has to be
+# modified by the function, `Y` is the current state, `p` is the cache (an
+# optional collection of auxiliary variables) and `t` the time. Tendencies in
+# `ClimaTimesteppers.jl` are in-place. In our case, we want the divergence of
+# the gradient (the Laplacian) of `my_var`.
+
+diverg = ClimaCore.Operators.WeakDivergence();
+grad = ClimaCore.Operators.Gradient();
+
+K = 3.0
+
+# Notice how we picked `WeakDivergence` instead of normal divergence.
+# TODO: Explain why
+
+function T_exp!(∂ₜY, Y, _, _)
+    ∂ₜY.my_var .= K .* diverg.(grad.(Y.my_var))
+    return nothing
+end
+
+# We do not specify boundary conditions here because we are on a sphere, where
+# the only meaningful boundary conditions are periodic.
+
+# Next, we move to the implicit tendency. This is much more involved to set up.
+# First, we need again divergence and gradient. However, this time we want them
+# for the vertical direction
+diverg_vert = ClimaCore.Operators.DivergenceC2F(;
+    bottom = ClimaCore.Operators.SetDivergence(0.0),
+    top = ClimaCore.Operators.SetDivergence(0.0),
+);
+grad_vert = ClimaCore.Operators.GradientF2C();
+
+# We choose `DivergenceC2F` and `GradientF2C` to implement the Laplacian. `F2C`
+# and `C2F` stand for `face-to-center` and `center-to-face`. The choice is
+# motivated so that we have a chain of operations that brings us back to faces.
+# We are working with faces so that we can cleanly impose boundary conditions.
+# The boundary condition we are setting here is null divergence at the top and
+# bottom of the domain.
+
+function actual_T_imp!(∂ₜY, Y, _, _)
```

Comment:
Because `T_imp!` is an `ODEFunction`, and `actual_T_imp!` is the function that applies the tendency.

Reply:
I think it might be a little more clear if you call this `T_imp!` and you call the other one something like `T_imp_wrapper` or `T_imp_and_jacobian` or `T_imp_ode_func`. If we decide to change this interface, we'll probably want to split the wrapper into `T_imp!` and `Wfact!` anyway.

Reply:
Sounds good!